src/node_messaging.cc

```diff
@@ -647,8 +761,18 @@ void MessagePort::OnMessage() {
       continue;
     }
 
-    Local<Function> emit_message = PersistentToLocal::Strong(emit_message_fn_);
     if (MakeCallback(emit_message, 1, &payload).IsEmpty()) {
+    reschedule:
+      if (!message_error.IsEmpty()) {
+        // This should become a `messageerror` event in the sense of the
+        // EventTarget API at some point.
+        Local<Value> argv[] = {
+          env()->messageerror_string(),
+          message_error
+        };
+        USE(MakeCallback(env()->emit_string(), arraysize(argv), argv));
```

Comment:
Would it make sense to emit `error` or `uncaughtException` if no one is listening to `messageerror`?

Reply:
(or a warning)

Reply:
@benjamingr I would prefer to drop the message in that case, rather than start emitting more events/raise it as an uncaught exception… I think if anything, a warning would be okay, but like you said below, this isn’t something that should really occur in normal flow (and I assume you meant to refer to that as *not* critical)?

So yeah, if you feel strongly about it, I’d add a warning, otherwise I’d probably say it’s not worth the extra complexity.

(Also, I don’t know if your LGTM counts as an approval of this PR?)